File: services/api/src/main/java/com/shiji/api/modules/meal/model/dto/response/LatestMealRecordResponse.java
package com.shiji.api.modules.meal.model.dto.response;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class LatestMealRecordResponse {

    private final Long mealRecordId;
    private final String mealType;
    private final LocalDateTime recordedAt;
    private final BigDecimal totalEstimatedCalories;
    private final Mood mood;

    @Getter
    @Builder
    public static class Mood {
        private final String emotionCode;
        private final String emotionName;
    }
}

File: apps/mobile/lib/core/network/models/latest_meal_record_response.dart
class LatestMealRecordResponse {
  LatestMealRecordResponse({
    required this.mealType,
    required this.recordedAt,
    required this.totalEstimatedCalories,
    required this.mood,
  });

  factory LatestMealRecordResponse.fromJson(Map<String, dynamic> json) {
    return LatestMealRecordResponse(
      mealType: (json['mealType'] as String?) ?? '',
      recordedAt: DateTime.parse(json['recordedAt'] as String),
      totalEstimatedCalories: (json['totalEstimatedCalories'] as num?)?.toDouble(),
      mood: json['mood'] is Map<String, dynamic>
          ? LatestMealMood.fromJson(json['mood'] as Map<String, dynamic>)
          : null,
    );
  }

  final String mealType;
  final DateTime recordedAt;
  final double? totalEstimatedCalories;
  final LatestMealMood? mood;
}

class LatestMealMood {
  LatestMealMood({
    required this.emotionCode,
    required this.emotionName,
  });

  factory LatestMealMood.fromJson(Map<String, dynamic> json) {
    return LatestMealMood(
      emotionCode: (json['emotionCode'] as String?) ?? '',
      emotionName: (json['emotionName'] as String?) ?? '',
    );
  }

  final String emotionCode;
  final String emotionName;
}
